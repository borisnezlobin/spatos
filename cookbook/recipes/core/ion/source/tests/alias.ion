alias piped = "sed s/A/B/ | sed s/B/C/"
alias logical = "echo LOGICAL1 && echo B"
alias piped_and_logical = "sed s/A/B/ | sed s/B/PIPED_AND_LOGICAL1/ && echo B"

# Alias contains only pipes
echo A | piped | sed s/C/PIPED/

# Alias containes logical operator
logical | sed s/B/LOGICAL2/

# Alias breaks pipeline into half by logical operator
echo A | piped_and_logical | sed s/B/PIPED_AND_LOGICAL2/

# Expanding multiple aliases in a singel pipeline
echo A | piped | sed s/C/PIPED_OK/ && logical | piped_and_logical | sed s/B/DONE/
