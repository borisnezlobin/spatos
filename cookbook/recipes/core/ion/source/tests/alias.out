PIPED
LOGICAL1
LOGICAL2
PIPED_AND_LOGICAL1
PIPED_AND_LOGICAL2
PIPED_OK
LOGICAL1
PIPED_AND_LOGICAL1
DONE
